Winget-AutoUpdate/functions/Get-WingetOutdatedApps.ps1

```diff
@@ -38,9 +38,22 @@ function Get-WingetOutdatedApps {
 
     # Now cycle in real package and split accordingly
     $upgradeList = @()
-    For ($i = $fl + 2; $i -lt $lines.Length - 1; $i++) {
+    For ($i = $fl + 2; $i -lt $lines.Length; $i++) {
         $line = $lines[$i]
-        if ($line) {
+        if ($line -and $line.StartsWith("-----")) {
+            #Get header line
+            $fl = $i - 1
+
+            #Get header titles
+            $index = $lines[$fl] -split '\s+'
+
+            # Line $fl has the header, we can find char where we find ID and Version
+            $idStart = $lines[$fl].IndexOf($index[1])
+            $versionStart = $lines[$fl].IndexOf($index[2])
+            $availableStart = $lines[$fl].IndexOf($index[3])
+       }
+       #(Alphanumeric | Literal . | Alphanumeric) - the only unique thing in common for lines with applications
+        if ($line -and ($line -match "\w\.\w")) {
```

Comment:
What if we put your regex check directly in the first "if" on line 43 ?

Reply:
I don't think so.
The new header has another char index where we find ID and version, so that must be compensated.
And, this way we take in account if theres a lot of different headers too.

Reply:
And, regarding alphanumeric (som have versions without dot, but Id have it):
WezTerm                                     wez.wezterm                         20221119-145034-4

Reply:
..or versions like:
unetbootin                                  unetbootin.unetbootin               702

Reply:
yes, but "\w\.\w" will match for wez.wezterm and unetbootin.unetbootin

Reply:
> The new header has another char index where we find ID and version, so that must be compensated. And, this way we take in account if theres a lot of different headers too.

I got it.

Reply:
ok, seams very good :)
maybe `if ($line -and $line.StartsWith("-----"))` could just be `if ($line.StartsWith("-----"))`
and `if ($line -and ($line -match "\w\.\w"))` just `if ($line -match "\w\.\w")` ?

Reply:
Make sense. After Where-Object { $_ } it's not needed again.